eidas_node/saml.py

```diff
@@ -111,11 +117,61 @@ def from_light_request(cls: Type[SAMLRequestType], light_request: LightRequest,
         SubElement(SubElement(root, Q_NAMES['saml2p:RequestedAuthnContext'], {'Comparison': 'minimum'}),
                    Q_NAMES['saml2:AuthnContextClassRef']).text = light_request.level_of_assurance.value
         # 8: AuthnRequestType <saml2p:Scoping> skipped
-        return cls(ElementTree(root), light_request.relay_state)
+        return cls(ElementTree(root), light_request.citizen_country_code, light_request.relay_state)
+
+    def create_light_request(self) -> LightRequest:
+        """
+        Convert SAML Request to Light Request.
+
+        :return: A Light Request.
+        :raise ValidationError: If the SAML Request cannot be parsed correctly.
+        """
+        request = LightRequest(requested_attributes=OrderedDict(),
+                               citizen_country_code=self.citizen_country_code,
+                               relay_state=self.relay_state)
+        root = self.document.getroot()
+        if root.tag != Q_NAMES['saml2p:AuthnRequest']:
+            raise ValidationError({get_element_path(root): 'Wrong root element: {!r}'.format(root.tag)})
+
+        request.id = root.attrib.get('ID')
+        request.provider_name = root.attrib.get('ProviderName')
+
+        for elm in root:
```

Comment:
This is far too complex and needs to be rewritten. It should be possible to use `find` from lxml.

Reply:
Ported to `find` and `findall`.